### PR TITLE
Add github action to automate deploy to github pages

### PR DESCRIPTION
**Pull request by mattborghi**
Creation of GitHub action that automatically deploys to github pages on PR merged and push to master.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  push:
+    branches: [master]
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [12.x]
```
Comment (german-e-mas) [inline .github/workflows/deploy.yml:17]: Aren't we using `14.x`?

Comment (mattborghi) [reply .github/workflows/deploy.yml:15]: I chose to keep the same version as the other action file. 

I am using `16,x` which is the latest LTS version.

https://github.com/ekumenlabs/telegram-tt/blob/3af7f1f89fb55f3689be01da33a822c47419b857/.github/workflows/tests.yml#L16

Comment (german-e-mas) [reply .github/workflows/deploy.yml:15]: Sounds good

**Review by german-e-mas — APPROVED** [review]
LGTM!

**Comment by german-e-mas** [comment]
Is there a way to disable `npm ci` for the failing checks?

**Pull request merged**